jwst/clean_flicker_noise/lib.py

```diff
@@ -459,67 +500,75 @@ def fit(self, return_fit=False, weight_fit=False):
 
         # Define which Fourier vectors to fit. For consistency with numpy, call this k.
-        k = np.arange(len(self.rfftfreq))[self.apodizer>0.].reshape((1,-1))
+        k = np.arange(len(self.rfftfreq))[self.apodizer > 0.0].reshape((1, -1))
 
         # Build the incomplete Fourier matrix
-        B = np.array(np.exp(2*np.pi*1J*m*k/self.n)/m.shape[0], dtype=np.complex64)
+        basis = np.array(np.exp(2 * np.pi * 1j * m * k / self.n) / m.shape[0], dtype=np.complex64)
 
         # Weighted NSClean fitting
         if weight_fit:
-
             # Build the weight matrix. Weight by the reciprocal of the local background
             # sample density in time. Roughly approximate the local density, P, using
             # the reciprocal of convolution by a Gaussian kernel.
-            _x = np.arange(self.n, dtype=np.float32) # x-values for building a 1-D Gaussian
-            _mu = np.float32(self.n//2+1) # Center point of Gaussian
-            _sigma = np.float32(self.weights_kernel_sigma) # Standard deviation of Gaussian
-            W = np.exp(-(_x-_mu)**2/_sigma**2/2)/_sigma/np.sqrt(2*np.pi) # Build centered Gaussian
-            FW = np.fft.rfft(np.fft.ifftshift(W)) # Forward FFT
-            _M = np.hstack((self.mask, np.zeros((self.ny,self.nloh), dtype=np.bool_))).flatten() # Add new line overhead to mask
-            with np.errstate(divide='ignore'):
-                P = 1/np.fft.irfft(np.fft.rfft(np.array(_M, dtype=np.float32)) * FW, self.n) # Compute weights
-            P = P[_M] # Keep only background samples
+            _x = np.arange(self.n, dtype=np.float32)  # x-values for building a 1-D Gaussian
+            _mu = np.float32(self.n // 2 + 1)  # Center point of Gaussian
+            _sigma = np.float32(self.weights_kernel_sigma)  # Standard deviation of Gaussian
+            _weight = (
+                np.exp(-((_x - _mu) ** 2) / _sigma**2 / 2) / _sigma / np.sqrt(2 * np.pi)
+            )  # Build centered Gaussian
+            _weight_fft = np.fft.rfft(np.fft.ifftshift(_weight))  # Forward FFT
+            _m = np.hstack(
+                (self.mask, np.zeros((self.ny, self.nloh), dtype=np.bool_))
+            ).flatten()  # Add new line overhead to mask
+            with np.errstate(divide="ignore"):
+                p_matrix = 1 / np.fft.irfft(
+                    np.fft.rfft(np.array(_m, dtype=np.float32)) * _weight_fft, self.n
+                )  # Compute weights
+            p_matrix = p_matrix[_m]  # Keep only background samples
 
             # NSClean's weighting requires the Moore-Penrose inverse of A = P*B.
             #     $A^+ = (A^H A)^{-1} A^H$
-            A = P.reshape((-1,1)) * B # P is diagonal. Hadamard product is most RAM efficient
-            AH = np.conjugate(A.transpose()) # Hermitian transpose of A
-            pinv_PB = np.matmul(np.linalg.inv(np.matmul(AH, A)), AH)
+            _a = (
+                p_matrix.reshape((-1, 1)) * basis
```

Comment:
similar comment here, it looks strange to have this operation split onto three lines. I suggest to move the comment instead